src/iam.ts

```diff
@@ -487,16 +492,27 @@ export class IAM extends IAMBase {
      *
      */
     async issuePublicClaim({ token, publicClaim }: { token?: string; publicClaim?: IPublicClaim }) {
-        if (this._issuerClaims) {
-            if (publicClaim) {
-                return this._issuerClaims.issuePublicClaim(publicClaim);
-            }
-            if (token) {
-                return this._issuerClaims.issuePublicClaim(token);
-            }
+        // Throw error when IssuerClaims is not initialized
+        if (!this._issuerClaims) {
+            throw new Error(ERROR_MESSAGES.CLAIMS_NOT_INITIALIZED);
+        }
+
+        // Throw error when there is no claim to issue
+        if (!publicClaim && !token) {
             throw new Error("unable to issue Public Claim");
         }
-        throw new Error(ERROR_MESSAGES.CLAIMS_NOT_INITIALIZED);
+
+        let issuedToken = "";
+        if (publicClaim) {
+            issuedToken = await this._issuerClaims.issuePublicClaim(publicClaim);
+        }
+        if (token && !publicClaim) {
+            issuedToken = await this._issuerClaims.issuePublicClaim(token);
+        }
+
+        this._cacheClient.saveIssuedToken({ issuedToken });
```

Comment:
I think this is adding the side-effect of saving the IssuedToken, right? I would add a boolean to the method params, like `saveToken` that is `false` by default that optionally saves the issuedToken